analyzers/src/SonarAnalyzer.Common/SymbolicExecution/SymbolicConstraint.cs

```diff
@@ -24,6 +24,7 @@ public abstract class SymbolicConstraint
     {
         public abstract SymbolicConstraint Opposite { get; }
         protected abstract string Name { get; }
+        public virtual bool PreserveOnFieldReset => false;
```

Comment:
How is it possible now that we don't have to override it in `LockConstraint`???

Let's add a specific test case to `LocksReleasedAllPaths.Monitor.Conditions.cs` to break it. Otherwise, the property would not make sense and I still believe it makes sense